### PR TITLE
CAMS-473 - Identify legacy and invalid divisions and filter where necessary

### PR DESCRIPTION
**Pull request by btposey**
Jira ticket: CAMS-473

# Purpose

Identify legacy and invalid divisions and filter where necessary

Remove legacy Islip division from transfer court list.

# Major Changes

* Added lists of invalid and legacy division codes.
* Decorated the division model with tags.
* Added INVALID and LEGACY tags to divisions in the lists.

# Testing/Validation

Manual testing of the API

### CODE REVIEW

File: backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ import { buildOfficeCode, getOfficeName } from '../../../use-cases/offices/offic
 
 const MODULE_NAME = 'OFFICES-GATEWAY';
 
+// Remove invalid divisions at the gateway rather than forcing the
+// more important use case code to include logic to remove them.
```
Comment (jamesobrooks) [inline backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts:18]: For legacy codes, I would say this is more of a business logic issue than a persistence issue, but we can leave it for an axis of change in the future to decide whether/how to treat them differently than the invalid ones.

File: backend/functions/lib/adapters/gateways/storage/local-storage-gateway.ts [hunk 2]
```diff
@@ -55,10 +61,33 @@ function getRoleMapping(): Map<string, CamsRole> {
...
+  });
+}
+
+function getUstpDivisionMeta(): Map<string, UstpDivisionMeta> {
+  if (!metaMapping) {
+    metaMapping = new Map<string, UstpDivisionMeta>();
+    addUstpDivisionMetaToMap(metaMapping, { isLegacy: true }, LEGACY_DIVISION_CODES);
+  }
+  return metaMapping;
+}
```
Comment (jamesobrooks) [inline backend/functions/lib/adapters/gateways/storage/local-storage-gateway.ts:84]: I am not seeing what this code is doing for us. It seems that we're never adding any codes to `LEGACY_DIVISION_CODES` so line 69 has nothing to iterate over.

File: backend/functions/lib/use-cases/offices/offices.ts [hunk 3]
```diff
@@ -17,8 +17,23 @@ const MODULE_NAME = 'OFFICES_USE_CASE';
...
+    const offices = await officesGateway.getOffices(context);
+
+    const storageGateway = getStorageGateway(context);
+    const metas = storageGateway.getUstpDivisionMeta();
+
+    offices.forEach((ustpOffice) => {
+      ustpOffice.groups.forEach((group) => {
+        group.divisions.forEach((division) => {
+          if (metas.has(division.divisionCode)) {
+            division.isLegacy = metas.get(division.divisionCode).isLegacy;
```
Comment (jamesobrooks) [inline backend/functions/lib/use-cases/offices/offices.ts:30]: With the current definition of UstpDivisionMeta, this assignment seems to violate the type:
```
export type UstpDivisionMeta = {
  isLegacy?: true;
};
```
So if `isLegacy` exists, it should be `true`, but we're potentially setting it to `false`.

File: backend/functions/lib/controllers/orders/orders.controller.test.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ import { mockCamsHttpRequest } from '../../testing/mock-data/cams-http-request-h
 import { ResponseBody } from '../../../../../common/src/api/response';
 import { NotFoundError } from '../../common-errors/not-found-error';
 import { BadRequestError } from '../../common-errors/bad-request';
+import * as crypto from 'crypto';
```
Comment (jamesobrooks) [inline backend/functions/lib/controllers/orders/orders.controller.test.ts:24]: Why is this added?

Comment (btposey) [reply backend/functions/lib/controllers/orders/orders.controller.test.ts:24]: This missing import was causing the test to fail for me locally.

File: backend/functions/lib/use-cases/offices/offices.ts (hunk 3 above)
Comment (btposey) [reply backend/functions/lib/use-cases/offices/offices.ts:30]: I intend for it to be absent if not applicable.

File: backend/functions/lib/adapters/gateways/storage/local-storage-gateway.ts (hunk 2 above)
Comment (btposey) [reply backend/functions/lib/adapters/gateways/storage/local-storage-gateway.ts:84]: I did have "070" in here, but Jon confirmed that the division had no legacy cases so I moved it to the "invalid" list in the gateway. I left the legacy logic in place because I'm pretty sure we will be adding legacy division codes in the near term.

File: backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts [hunk 5]
```diff
@@ -14,6 +14,11 @@ import { buildOfficeCode, getOfficeName } from '../../../use-cases/offices/offic
...
+const INVALID_DIVISION_CODES = ['070', '990', '991', '992', '993', '994', '995', '996', '999'];
```
Comment (fmaddenflx) [inline backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts:19]: Curious to know where we got this list.

Comment (btposey) [reply backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts:19]: From our analysis of divisions exported from DXTR. Some are obvious. Then 070 is for Islip. We have more to come in the future as the analysis is completed.
